server/middleware/validation.js

```diff
@@ -17,13 +17,38 @@ const validationErrors = {
     isValidWithinDate: label => `${label} must be within the last ${VALID_DAYS_RANGE} days`,
     validCaseReference: () => 'Case reference is not valid',
     contributionsFulfilled: () => 'Case contributions have to be completed or cancelled',
+    approvalsFulfilled: () => 'Case approvals have to be completed or cancelled',
     oneOf: () => 'Select at least one option',
     isValidMonth: label => `${label} must contain a real month`,
     isBeforeMaxYear: label => `${label} must be before ${MAX_ALLOWABLE_YEAR}`,
     isAfterMinYear: label => `${label} must be after ${MIN_ALLOWABLE_YEAR}`,
     isValidDay: label => `${label} must contain a real day`,
 };
 
+const requestsFulfilled = ( value, message, statusField, cancelledValue, completeValue, defaultError ) => {
+    let valid = true;
+    try {
+        const contributions = JSON.parse(value);
+
+        if (Array.isArray(contributions)) {
+            const result = contributions.filter(contribution => {
+                const contributionStatus = contribution.data[statusField];
+                return (!(contributionStatus === cancelledValue || contributionStatus === completeValue));
+            });
+
+            if (result.length !== 0) {
+                valid = false;
+            }
+        } else {
+            valid = false;
+        }
+    } catch (error) {
+        valid = false;
```

Comment:
Any reason why we aren't logging out the error here? Seems to me the likely culprit would be malformed JSON response which is an error we would want to know about.

Reply:
I've now added a logging statement here.

Reply:
great thanks